content_resolver/analyzer.py

```diff
@@ -15,6 +17,315 @@ def pkg_placeholder_name_to_nevr(placeholder_name):
     return placeholder_id
 
 
+###############################################################################
+### Standalone functions for ProcessPoolExecutor ############################
+###############################################################################
+
+def _get_build_deps_from_a_root_log_standalone(root_log):
+    """
+    Standalone version of _get_build_deps_from_a_root_log for ProcessPoolExecutor.
+    Must not reference class methods or instance variables.
+    """
+    required_pkgs = []
+
+    # The individual states are nicely described inside the for loop.
+    # They're processed in order
+    state = 0
+    
+    for file_line in root_log.splitlines():
+
+        # 0/
+        # parts of the log I don't really care about
+        if state == 0:
+
+            # The next installation is the build deps!
+            # So I start caring. Next state!
+            if "'builddep', '--installroot'" in file_line:
+                state += 1
+        
+
+        # 1/
+        # getting the "already installed" packages to the list
+        elif state == 1:
+
+            # "Package already installed" indicates it's directly required,
+            # so save it.
+            # DNF5 does this after "Repositories loaded" and quotes the NVR;
+            # DNF4 does this before "Dependencies resolved" without the quotes.
+            if "is already installed." in file_line:
+                # Format: "Package bash-5.1.0-1.fc34.x86_64 is already installed."
+                words = file_line.split()
+                if len(words) >= 4 and words[0] == "Package" and words[-2] == "already":
+                    pkg_nevr = words[1].strip('"')
+                    pkg_name = pkg_nevr.rsplit("-", 2)[0]
```

Comment:
Keep the original parser:
```suggestion
                pkg_name = file_line.split()[3].strip('"').rsplit("-",2)[0]
```